docs/source/add_new_model.md

```diff
@@ -0,0 +1,573 @@
+the 🤗Transformers implementation or you need to transpose one of the checkpoint weights.
+
+This step should be iterated with the previous step until all weights of the checkpoint are correctly loaded in the Transformers model.
+Having correctly loaded the checkpoint into the 🤗Transformers implementation, you can then save the model under a folder of your choice `/path/to/converted/checkpoint/folder` that should include both a `pytorch_model.bin` file and a `config.json` file.
+
+#### Implement the forward pass
+
+Having managed to correctly load the pretrained weights into the 🤗Transformers implementation, you should now make sure that the forward pass is correctly implemented. In *Get familiar with the original repository.*, you have already created a script
+that runs a forward pass of the model using the original repository. Now you should write an analogous script using the 
+🤗Transformers implementation instead of the original one. It should look as follows:
+
+```python
+model = brand_new_bertModel.from_pretrained(/path/to/converted/checkpoint/folder)
+input_ids = ... # the exact same vector of input ids in PyTorch as those used in the *Get familiar with the original repository.* section
+output = model(input_ids).last_hidden_states
+```
+
+It is very likely that the 🤗Transformers implementation and the original model implementation don't give the exact same output the very first time or that the forward pass throws an error. Don't be disappointed - it's expected! First, you should make sure that the forward pass doesn't throw any errors. It often happens that the wrong dimensions are used leading to a `Dimensionality mismatch` error or that the wrong data type object is used,
+*e.g.* `torch.long` instead of `torch.float32`. Don't hesitate to ask the Hugging Face team for help, if you don't manage to solve certain errors.
+
+The final part to make sure the 🤗Transformers implementation works correctly is to ensure that the outputs are equivalent to a precision of `1e-3`.
+First, you should ensure that the output shapes are identical, *i.e.* `outputs.shape` should yield the same value for the script of the 🤗Transformers implementation and the original implementation. Next, you should make sure that the output values are identical as well. This one of the most difficult parts of adding a new model. Common mistakes why the outputs are not identical are:
+
+- Some layers were not added, *i.e.* an `activation` layer was not added, or the residual connection was forgotten
+- The word embedding matrix was not tied
+- The wrong positional embeddings are used because the original implementation uses on offset
+- Dropout is applied during the forward pass. To fix this make sure `model.training is False` and that no dropout layer is falsely activated during the 
+  forward pass, *i.e.* pass `self.training` to [PyTorch's functional dropout](https://pytorch.org/docs/stable/nn.functional.html?highlight=dropout#torch.nn.functional.dropout)
```

Comment:
```suggestion
- Some layers were not added, *i.e.* an `activation` layer was not added, or the residual connection was forgotten.
- The word embedding matrix was not tied.
- The wrong positional embeddings are used because the original implementation uses on offset.
- Dropout is applied during the forward pass. To fix this make sure `model.training is False` and that no dropout layer is falsely activated during the 
  forward pass, *i.e.* pass `self.training` to [PyTorch's functional dropout](https://pytorch.org/docs/stable/nn.functional.html?highlight=dropout#torch.nn.functional.dropout).
```